src/main/java/org/checkerframework/checker/noliteral/NoLiteralVisitor.java

```diff
@@ -0,0 +1,230 @@
+package org.checkerframework.checker.noliteral;
+
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.Tree;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.ExecutableElement;
+import org.checkerframework.checker.noliteral.qual.MaybeConstant;
+import org.checkerframework.checker.noliteral.qual.NonConstant;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.common.basetype.BaseTypeVisitor;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedExecutableType;
+import org.checkerframework.framework.type.AnnotatedTypeParameterBounds;
+import org.checkerframework.framework.type.TypeHierarchy;
+import org.checkerframework.javacutil.ElementUtils;
+
+/**
+ * This visitor has a lot of code in it that permits this type system to operate as expected with
+ * the bottom type as its default. Most of that code is either from or inspired by AWS' Data
+ * Classification Checker (https://github.com/awslabs/data-classification-checker).
+ */
+public class NoLiteralVisitor extends BaseTypeVisitor<NoLiteralAnnotatedTypeFactory> {
+  /**
+   * @param checker the type-checker associated with this visitor (for callbacks to {@link
+   *     TypeHierarchy#isSubtype})
+   */
+  public NoLiteralVisitor(BaseTypeChecker checker) {
+    super(checker);
+  }
+
+  /**
+   * Overrides the default lower bound for exception parameters. By default, this is top. This
```

Comment:
I don't know whether the "This" at the end of the line (the start of the third sentence) refers to the first sentence or the second.  Please clarify the writing.